plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseTestHiveOnDataLake.java

```diff
@@ -1910,6 +1910,62 @@ public static Object[][] s3SelectFileFormats()
         };
     }
 
+    @Test
+    public void testTrinoDropSchemaCascade()
+    {
+        String schemaName = "test_drop_schema_cascade_" + randomNameSuffix();
+        String tableName = "test_table" + randomNameSuffix();
+        String trinoViewName = "test_trino_view" + randomNameSuffix();
+        String hiveViewName = "test_hive_view" + randomNameSuffix();
+        try {
+            assertUpdate("CREATE SCHEMA hive.%2$s WITH (location='s3a://%1$s/%2$s')".formatted(bucketName, schemaName));
+            assertUpdate("CREATE TABLE " + schemaName + "." + tableName + "(a INT)");
+            assertUpdate("CREATE VIEW " + schemaName + "." + trinoViewName + " AS SELECT 1 a");
+            hiveMinioDataLake.getHiveHadoop().runOnHive("CREATE VIEW " + schemaName + "." + hiveViewName + " AS SELECT * FROM " + schemaName + "." + tableName);
+
+            assertThat(computeActual("SHOW SCHEMAS").getOnlyColumnAsSet()).contains(schemaName);
+            assertThat(metastoreClient.getDatabase(schemaName)).isNotEmpty();
+            assertThat(metastoreClient.getAllTables(schemaName)).isNotEmpty();
+            assertThat(hiveMinioDataLake.getMinioClient().listObjects(bucketName, schemaName)).isNotEmpty();
+
+            assertUpdate("DROP SCHEMA " + schemaName + " CASCADE");
+
+            assertThat(computeActual("SHOW SCHEMAS").getOnlyColumnAsSet()).doesNotContain(schemaName);
+            assertThat(metastoreClient.getDatabase(schemaName)).isEmpty();
+            assertThat(metastoreClient.getAllTables(schemaName)).isEmpty();
+            assertThat(hiveMinioDataLake.getMinioClient().listObjects(bucketName, schemaName)).isEmpty();
+        }
+        finally {
+            assertUpdate("DROP TABLE IF EXISTS " + schemaName + "." + tableName);
+            assertUpdate("DROP VIEW IF EXISTS " + schemaName + "." + trinoViewName);
+            hiveMinioDataLake.getHiveHadoop().runOnHive("DROP VIEW IF EXISTS " + schemaName + "." + hiveViewName);
+            assertUpdate("DROP SCHEMA IF EXISTS " + schemaName);
+        }
+    }
+
+    @Test
+    public void testHiveDropSchemaCascade()
+    {
+        String schemaName = "test_drop_schema_cascade_" + randomNameSuffix();
```

Comment:
This method tests functionality which relates fully to Hive.
Why do we want it here?

Reply:
This test is helpful to understand Hive behavior and catch their future change. Left a code comment.